functions/Get-DbaMsdtc.ps1

```diff
@@ -90,77 +94,73 @@ Get DTC status for the computer srv0042 and show in a grid view
+			New-PSDrive -Name HKCR -PSProvider Registry -Root HKEY_CLASSES_ROOT | Out-Null
 			Get-ItemProperty -Path HKCR:\CID\*\Description |
 			Select-Object @{ l = 'Data'; e = { $_.'(default)' } }, @{ l = 'CID'; e = { $_.PSParentPath.split('\')[-1] } }
-			Remove-PSDrive -Name HKCR | Out-Null;
+			Remove-PSDrive -Name HKCR | Out-Null
 		}
 	}
-	PROCESS
+PROCESS
 	{
-		ForEach ($Computer in $Computername)
-		{
-			$dtcservice = $null
-			try
-			{
-				Write-Verbose "Getting DTC on $computer via WinRM"
-				$dtcservice = Get-Ciminstance -ComputerName $Computer -Query $query -ErrorAction Stop |
-				Select-Object PSComputerName, DisplayName, State, status, StartMode, StartName
-			}
-			catch
-			{
-				try
-				{
-					Write-Verbose "Failed To get DTC via WinRM. Getting DTC on $computer via DCom"
-					$SessionParams = @{ }
-					$SessionParams.ComputerName = $Computer
-					$SessionParams.SessionOption = (New-CimSessionOption -Protocol Dcom)
-					$Session = New-CimSession @SessionParams
-					$dtcservice = Get-Ciminstance -CimSession $Session -Query $query -ErrorAction Stop |
-					Select-Object PSComputerName, DisplayName, State, status, StartMode, StartName
-				}
-				catch
-				{
-					Write-Warning "Can't connect to CIM on $Computer"
-				}
-			}
-			$reg = $CIDs = $null
-			$CIDHash = @{ }
-			try
-			{
-				Write-Verbose "Getting Registry Values on $Computer"
-				$reg = Invoke-Command -ComputerName $Computer -ScriptBlock $DTCSecurity -ErrorAction Stop
-				$CIDs = Invoke-Command -ComputerName $Computer -ScriptBlock $DTCCIDs -ErrorAction Stop
-				foreach ($key in $CIDs) { $CIDHash.Add($key.Data, $key.CID) }
-			}
-			catch
-			{
-				Write-Warning "Can't connect to registry on $Computer"
-			}
-			if ($dtcservice)
-			{
-				[PSCustomObject]@{
-					ComputerName = $dtcservice.PSComputerName
-					DTCServiceName = $dtcservice.DisplayName
-					DTCServiceState = $dtcservice.State
-					DTCServiceStatus = $dtcservice.Status
-					DTCServiceStartMode = $dtcservice.StartMode
-					DTCServiceAccount = $dtcservice.StartName
-					DTCCID_MSDTC = $CIDHash['MSDTC']
-					DTCCID_MSDTCUIS = $CIDHash['MSDTCUIS']
-					DTCCID_MSDTCTIPGW = $CIDHash['MSDTCTIPGW']
-					DTCCID_MSDTCXATM = $CIDHash['MSDTCXATM']
-					networkDTCAccess = $reg.networkDTCAccess
-					networkDTCAccessAdmin = $reg.networkDTCAccessAdmin
-					networkDTCAccessClients = $reg.networkDTCAccessClients
-					networkDTCAccessInbound = $reg.networkDTCAccessInbound
-					networkDTCAccessOutBound = $reg.networkDTCAccessOutBound
-					networkDTCAccessTip = $reg.networkDTCAccessTip
-					networkDTCAccessTransactions = $reg.networkDTCAccessTransactions
-					XATransactions = $reg.XATransactions
-				}
-			}
-		}
+    ForEach ($Computer in $Computername)
+    {
+      $dtcservice = $null
+      Write-Verbose "$functionName - Getting DTC on $computer via WinRM"
+      $dtcservice = Get-Ciminstance -ComputerName $Computer -Query $query

```

Comment:
yes, error handling needs to be addressed in general.
I used to put -ErrorAction Silentlycontinue everywhere, but then I realised it would be better not to, and left it out without putting in a better way.

When using try catch around a variable assignment, we have lots of cases the catch is useless, since the assignment succeeds, even when assigning $null. So we replaced that with Silently continue and then an if else, checking if there is something meaningful in the variable.